### PR TITLE
Fix Onboarding Gradient

### PR DESCRIPTION
**Pull request by pnor**
## Overview

Fixed an issue where the fade gradients weren't matching up with the background.
Also consolidated the logic that controlled fade colors, so the OnboardingInterestsVC and GroupsVC wasn't reusing copies of the same code.

## Changes Made

-  Changed the background color of the gradients to match the image
- Create `FadeTableView` which wraps the tableview in a `UIView` that adds the fade effects
- Refactor other VCs to use new view

## Test Coverage

Ran on iOS simulator with iPhone 11 Pro Max

## Next Steps (delete if not applicable)

Right now, the color determining the fade is just a constant from `UIColors+Shared`. It'd be nice if it could be based off the background of the VC so the fade always works, but since the background is an image, I couldn't do that

## Related PRs or Issues (delete if not applicable)

#45 

## Screenshots (delete if not applicable)

<details>

  <summary>New Gradient Effect</summary>
  
  
image
</details>

<details>
<summary>Entire Phone</summary>
image
</details>

### CODE REVIEW

File: CoffeeChat/Controller/GroupsViewController.swift [hunk 1]
```diff
@@ -64,20 +59,18 @@ class GroupsViewController: UIViewController {
...
-        tableView.separatorStyle = .none
-        tableView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 30, right: 0)
-        view.addSubview(tableView)
-        view.addSubview(topFadeView)
-        view.addSubview(bottomFadeView)
+        fadeTableView.tableView.delegate = self
+        fadeTableView.tableView.dataSource = self
+        fadeTableView.tableView.backgroundColor = .none
+        fadeTableView.tableView.register(OnboardingTableViewCell.self, forCellReuseIdentifier: OnboardingTableViewCell.reuseIdentifier)
+        fadeTableView.tableView.isScrollEnabled = true
```
Comment (danielVebman) [inline CoffeeChat/Controller/GroupsViewController.swift:66]: Isn't this true by default?

File: CoffeeChat/Controller/OnboardingInterestsViewController.swift [hunk 2]
```diff
@@ -60,21 +55,19 @@ class OnboardingInterestsViewController: UIViewController {
...
-        tableView.showsVerticalScrollIndicator = false
-        tableView.separatorStyle = .none
-        tableView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 30, right: 0)
-        view.addSubview(tableView)
-        view.addSubview(topFadeView)
-        view.addSubview(bottomFadeView)
+        fadeTableView.tableView.delegate = self
+        fadeTableView.tableView.dataSource = self
+        fadeTableView.tableView.register(OnboardingTableViewCell.self, forCellReuseIdentifier: OnboardingTableViewCell.reuseIdentifier)
+        fadeTableView.tableView.isScrollEnabled = true
```
Comment (danielVebman) [inline CoffeeChat/Controller/OnboardingInterestsViewController.swift:61]: Ditto ^

**Review by danielVebman — APPROVED** [review]

File: CoffeeChat/Cells/OnboardingTableViewCell.swift [hunk 3]
```diff
@@ -9,7 +9,6 @@
 import UIKit
 
 class OnboardingTableViewCell: UITableViewCell {
-
```
Comment (ywsang) [inline CoffeeChat/Cells/OnboardingTableViewCell.swift:12]: do you know what the convention is for new line or not after class declaration in the other files? nitpicky, but it's nice to keep it consistent

File: CoffeeChat/Controller/GroupsViewController.swift (hunk 1 above)
Comment (ywsang) [reply CoffeeChat/Controller/GroupsViewController.swift:66]: ^^ i believe it is

File: CoffeeChat/Controller/OnboardingInterestsViewController.swift [hunk 4]
```diff
@@ -60,21 +55,19 @@ class OnboardingInterestsViewController: UIViewController {
...
+        fadeTableView.tableView.clipsToBounds = true
+        fadeTableView.tableView.backgroundColor = .none
+        fadeTableView.tableView.allowsMultipleSelection = true
+        fadeTableView.tableView.bounces = false
+        fadeTableView.tableView.showsHorizontalScrollIndicator = false
+        fadeTableView.tableView.showsVerticalScrollIndicator = false
+        fadeTableView.tableView.separatorStyle = .none
+        fadeTableView.tableView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 30, right: 0)
+        view.addSubview(fadeTableView)
```
Comment (ywsang) [inline CoffeeChat/Controller/OnboardingInterestsViewController.swift:70]: kinda piggybacking off of daniel's comment. looking at the two uses of the `fadeTableView` it seems like all of these configuration options are the same, so i think it makes sense to do default styling in the `FadeTableView`, that can be overridden if needed

**Review by ywsang — APPROVED** [review]
nice idea to factor the fade table view out into a component! just a comment about the styling

**Review by camruss — APPROVED** [review]
clean.

File: CoffeeChat/SceneDelegate.swift [hunk 5]
```diff
@@ -47,6 +47,8 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
             let navigationController = UINavigationController(rootViewController: LoginViewController())
             window.rootViewController = navigationController
         }
+        window.rootViewController = UINavigationController(rootViewController: OnboardingPageViewController(transitionStyle: UIPageViewController.TransitionStyle.scroll, navigationOrientation: UIPageViewController.NavigationOrientation.horizontal)
```
Comment (lucyxubroad) [inline CoffeeChat/SceneDelegate.swift:50]: What is this again? Was it just for testing purposes?

**Review by lucyxubroad — COMMENTED** [review]
Yay, good thinking making a new component haha cause we do use the fade table view a lot

**Comment by pnor** [comment]
Aa just realized I forgot to name it `phillip/onboarding-gradient` instead of `onboarding-gradient` 😫

**Pull request merged**